pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/CommittingSegmentDescriptor.java

```diff
@@ -16,29 +16,67 @@
 
 package com.linkedin.pinot.controller.helix.core.realtime.segment;
 
-import com.linkedin.pinot.common.metadata.segment.LLCRealtimeSegmentZKMetadata;
+import com.linkedin.pinot.common.protocols.SegmentCompletionProtocol;
 
 
 /**
  * Class to hold properties of the committing segment
  */
 public class CommittingSegmentDescriptor {
-  private LLCRealtimeSegmentZKMetadata _committingSegmentZkMetadata;
-  private long _committingSegmentSizeBytes;
+  private String _segmentName;
+  private long _segmentSizeBytes;
+  private String _segmentLocation;
+  private long _nextOffset;
 
-  public LLCRealtimeSegmentZKMetadata getCommittingSegmentZkMetadata() {
-    return _committingSegmentZkMetadata;
+  public static CommittingSegmentDescriptor fromSegmentCompletionReqParams(
```

Comment:
Or, you can have a constructor that takes RequestParams in the argument and copies over. That may be the only constructor we will need?